### PR TITLE
Refactor extension checkers

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/conversion/IconPackConversionViewModel.kt [hunk 1]
```diff
@@ -192,7 +194,7 @@ class IconPackConversionViewModel(
     }
 
     private fun List<Path>.processFiles() {
-        val paths = filter { it.isRegularFile() && (it.extension == "xml" || it.extension == "svg") }
+        val paths = filter { it.isRegularFile() && (it.isXml || it.isSvg) }
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/conversion/IconPackConversionViewModel.kt:197]: Checks lowercase and uppercase both for now.

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/conversion/util/FileToPainter.kt [hunk 2]
```diff
@@ -4,15 +4,17 @@ import androidx.compose.ui.graphics.painter.Painter
...
 import kotlin.io.path.extension
 import kotlin.io.path.name
 import kotlin.io.path.readText
 
-fun Path.toPainterOrNull(): Painter? = when (extension) {
-    "svg" -> svgToPainter()
-    "xml" -> xmlToPainter()
+fun Path.toPainterOrNull(): Painter? = when {
+    isSvg -> svgToPainter()
+    isXml -> xmlToPainter()
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/conversion/util/FileToPainter.kt:17]: Same as above.

File: components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/IconType.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    companion object {
+        fun from(extension: String): IconType? = when {
+            extension.isSvg -> SVG
+            extension.isXml -> XML
+            else -> null
+        }
+    }
+}
+
+inline val String?.isSvg: Boolean get() = equals(other = IconType.SVG.extension, ignoreCase = true)
```
Comment (egorikftp) [inline components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/IconType.kt:21]: Is it really should be nullable?)

Comment (Goooler) [reply components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/IconType.kt:21]: Simplify for `extension != null && (extension.isSvg() || extension.isXml())`.

**Pull request merged**